sdk/cosmos/azure-cosmos/azure/cosmos/_execution_context/aio/base_execution_context.py

```diff
@@ -141,7 +146,12 @@ async def _fetch_items_helper_no_retries(self, fetch_function):
 
     async def _fetch_items_helper_with_retries(self, fetch_function):
         async def callback():
-            return await self._fetch_items_helper_no_retries(fetch_function)
+            try:
+                return await self._fetch_items_helper_no_retries(fetch_function)
+            except exceptions.CosmosHttpResponseError as e:
+                if e.status_code == 429:
+                    self._has_started = False
```

Comment:
This is to check if we are running into 429s while testing, an attempt to reset `self._has_started` back to false as mentioned in PR 35022